### PR TITLE
PR: support minSelectedChoices for ranking(selectToRank) and checkboxes

### PR DESCRIPTION
**Pull request by dmitrykurmanov**
see https://github.com/surveyjs/private-tasks/issues/242

### CODE REVIEW

**Review by RomanTsukanov — APPROVED** [review]

File: src/question_text.ts [hunk 1]
```diff
@@ -257,7 +248,22 @@ export class QuestionTextModel extends QuestionTextBase {
       ); };
       errors.push(maxError);
     }
+
+    var name = this.name;
```
Comment (andrewtelnov) [inline src/question_text.ts:252]: const valName = this.getValidatorTitle();

File: src/question_text.ts [hunk 2]
```diff
@@ -257,7 +248,22 @@ export class QuestionTextModel extends QuestionTextBase {
...
+    var emailValidator = new EmailValidator();
+    if (
+      this.inputType === "email" &&
+      !this.validators.some((v) => v.getType() === "emailvalidator") &&
+      emailValidator.validate(this.value, name)
```
Comment (andrewtelnov) [inline src/question_text.ts:257]: Return the ValidatorResult class { value, error }. error can be null:
```js
export class ValidatorResult {
  constructor(public value: any, public error: SurveyError = null) {}
}

```

File: src/validator.ts [hunk 3]
```diff
@@ -38,7 +38,7 @@ export class SurveyValidator extends Base {
   get locText(): LocalizableString {
     return this.getLocalizableString("text");
   }
-  protected getErrorText(name: string): string {
+  public getErrorText(name: string): string {
```
Comment (andrewtelnov) [inline src/validator.ts:41]: Don't need to make it public you don't need to create the error, validator.validate()

File: src/question_text.ts [hunk 4]
```diff
@@ -249,18 +249,21 @@ export class QuestionTextModel extends QuestionTextBase {
...
+      !this.validators.some((v) => v.getType() === "emailvalidator")
     ) {
-      const maxError = new CustomError(
-        emailValidator.getErrorText(name),
-        this
-      );
-      errors.push(maxError);
+      const validateResult = emailValidator.validate(this.value, valName);
+
+      if (!!validateResult) {
```
Comment (andrewtelnov) [inline src/question_text.ts:260]: if (!!validateResult && !!validateResult.error) {
 errors.push(validateResult.error);
}

**Review by andrewtelnov — APPROVED** [review]

**Pull request merged**